tss-esapi/src/structures/buffers/public.rs

```diff
@@ -1,507 +1,168 @@
-    ) -> Self {
-        self.name_hashing_algorithm = Some(name_hashing_algorithm);
-        self
-    }
+impl Deref for PublicBuffer {
+    type Target = Vec<u8>;
 
-    /// Adds the auth policy for the [Public] structure
-    /// to the builder
-    pub fn with_auth_policy(mut self, auth_policy: &Digest) -> Self {
-        self.auth_policy = Some(auth_policy.clone());
-        self
+    fn deref(&self) -> &Self::Target {
+        &self.0
     }
+}
 
-    /// Adds the RSA parameters for the [Public] structure
-    /// to the builder.
-    ///
-    /// # Details
-    /// This is required if the public algorithm is set to
-    /// [Rsa][`crate::interface_types::algorithm::PublicAlgorithm::Rsa].
-    pub fn with_rsa_parameters(mut self, rsa_parameters: PublicRsaParameters) -> Self {
-        self.rsa_parameters = Some(rsa_parameters);
-        self
-    }
+impl TryFrom<Vec<u8>> for PublicBuffer {
+    type Error = Error;
 
-    /// Adds the RSA unique identifier for the [Public] structure
-    /// to the builder.
-    ///
-    /// # Details
-    /// This is required if the public algorithm is set to
-    /// [Rsa][`crate::interface_types::algorithm::PublicAlgorithm::Rsa].
-    ///
-    /// The unique identifier is the public key.
-    pub fn with_rsa_unique_identifier(mut self, rsa_unique_identifier: &PublicKeyRsa) -> Self {
-        self.rsa_unique_identifier = Some(rsa_unique_identifier.clone());
-        self
+    fn try_from(bytes: Vec<u8>) -> Result<Self> {
+        if bytes.len() > Self::MAX_SIZE {
+            error!("Error: Invalid Vec<u8> size(> {})", Self::MAX_SIZE);
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
+        Ok(PublicBuffer(bytes))
     }
+}
 
-    /// Adds the keyed hash parameters for the [Public] structure
-    /// to the builder.
-    ///
-    /// # Details
-    /// This is required if the public algorithm is set to
-    /// [KeyedHash][`crate::interface_types::algorithm::PublicAlgorithm::KeyedHash].
-    pub fn with_keyed_hash_parameters(
-        mut self,
-        keyed_hash_parameters: PublicKeyedHashParameters,
-    ) -> Self {
-        self.keyed_hash_parameters = Some(keyed_hash_parameters);
-        self
-    }
+impl TryFrom<&[u8]> for PublicBuffer {
+    type Error = Error;
 
-    /// Adds the keyed hash unique identifier for the [Public] structure
-    /// to the builder.
-    ///
-    /// # Details
-    /// This is required if the public algorithm is set to
-    /// [KeyedHash][`crate::interface_types::algorithm::PublicAlgorithm::KeyedHash].
-    pub fn with_keyed_hash_unique_identifier(
-        mut self,
-        keyed_hash_unique_identifier: &Digest,
-    ) -> Self {
-        self.keyed_hash_unique_identifier = Some(keyed_hash_unique_identifier.clone());
-        self
+    fn try_from(bytes: &[u8]) -> Result<Self> {
+        if bytes.len() > Self::MAX_SIZE {
+            error!("Error: Invalid &[u8] size(> {})", Self::MAX_SIZE);
```

Comment:
:thinking: why not attach `bytes.len()` to the error message such as:
```rust
error!("Error: Invalid &[u8] size: {} > {}", bytes.len(), Self::MAX_SIZE);
```

Reply:
Fixed